src/test/app/LedgerLoad_test.cpp

```diff
@@ -0,0 +1,221 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <test/support/jtx.h>
+#include <test/support/jtx/Env.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/SField.h>
+#include <boost/scope_exit.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string.hpp>
+#include <fstream>
+
+namespace ripple {
+
+class LedgerLoad_test : public beast::unit_test::suite
+{
+    Json::Value jvLedger_;
+    Json::Value jvHashes_;
+    boost::filesystem::path ledgerFile_;
+    boost::filesystem::path dbPath_;
+
+    auto ledgerConfig(std::string const& ledger, Config::StartUpType type)
+    {
+        auto p = std::make_unique<Config>();
+        test::setupConfigForUnitTests(*p);
+        p->START_LEDGER = ledger;
+        p->START_UP = type;
+        p->legacy("database_path", dbPath_.string());
```

Comment:
idk about all that, the rules for test code are more relaxed compared to production code. We should not worry about "accidental" calls. For unit tests, we prioritize convenience and compactness of notation over robustness